app/controllers/link_controller.rb

```diff
@@ -0,0 +1,63 @@
+require 'net/https'
+require 'uri'
+require 'json'
+require_relative '../../models/link'
+require_relative '../../app/controllers/application_controller'
+require_relative '../../models/category'
+require_relative '../../services/bing_service'
+require 'warden'
+
+class LinkController < ApplicationController
+  bing_search_api = BingService::BingWebSearchApi.new
+
+  get '/category_links/:category_id' do
+    redirect '/login' unless current_user
+    @current_user = current_user
+
+    term = @current_user.categories.find_by(id: params[:category_id].to_i)
```

Comment:
When searching using `primary key`, prefer to use `find` rather than `find_by` or `where`

```ruby
 term = @current_user.categories.find(params[:category_id])
```